### PR TITLE
Feature flags fetch fix

### LINKED ISSUE #1486: “Total Value” section not displayed on Home screen  when user update Valora application from previous build to latest build with logged in account
**Frequency:** 100%
**Repro on build version:** Android internal build V1.22.1
**Does Not Repro Version:** Test Flight Build V1.22.1
**Repro on devices:** One Plus 7T(11.0), Google Pixel 2XL (11.0), Samsung Galaxy F41  (11.0), Redmi note 8 (9.0)
**Does Not Repro on devices:** iPhone 12(15.0), iPhone SE (13.6.1), iPhone SE (14.2)
**Testing Account:** (680) 333 1978/[Recovery Phrase](https://docs.google.com/spreadsheets/d/1gXY1I4PBkT9f1uB3zxWgUUJJ1zjVeJcqcWLjiIls1f0/edit#gid=0&range=B11)

**Pre-condition:** Make sure that user account should have some balances

**Repro Steps:**
1. Launch the previous build (Play store build V1.21.0 here) & restore the account with the above credentials
2. Now observe balances from the Hamburger menu
3. Now update Valora application to the latest build (Android internal build V1.22.1 here)
4. Now redirect to the Home screen of an application
5. Observe that,

**Current Behavior:** “Total Value” section not displayed on Home screen & user not able to see the total sum of all balances on Home screen

**Expected Behavior:** User should be able to see the total sum of all balances in local currency with “Total value” section on the home screen 

**Investigation:** 
1. Also we have observed the same issue if we restore the account with no balances & then update the app to the latest version 
2. Also we have observed that user all balances displayed in Hamburger Menu instead of Home screen

**Impact:** Bad user experience as a user not able to see balances on Home screen after upgrading Valora application from previous to latest build

**Attachment:** [TotalValueSectionNotDisplayedAfterUpdating.mp4](https://drive.google.com/file/d/1BYty0fbFezLW2fxbrBdE-tLZORbBcC9r/view?usp=sharing)

### PR DESCRIPTION
**Pull request by gnardini**
### Description

The two main files of these changes are `packages/mobile/src/app/saga.ts` and `packages/mobile/src/firebase/firebase.ts`.
There are a few changes going on:
- Added a timeout for fetching remote feature flags. It was getting stuck otherwise so if the first request hanged because of a connection issue or otherwise it would never complete.
- Only update the `lastFetched` variable if flags are fetched correctly.
- We were only updating feature flags if new ones were fetched from remote. This caused issues when resetting Valora because there were no new feature flags but we were deleting the cache, so the user would be stuck with the defaults until a new feature flag was added or changed. Changed it to always emit the values from feature flags, even if nothing changed. No harm in doing so.

### Other changes

- Updated default values for some feature flags.
- Removed `shortVerificationCodesEnabled` feature flag

### Tested

Manually. I couldn't quite reproduce some of the reported errors on old versions anyhow, so I'd appreciate a good test on this from the QA team.

### How others should test

Try to reproduce errors where old feature flags were showing.

### Related issues

- Fixes #1486 and #1497

### Backwards compatibility

N/A

### CODE REVIEW

File: packages/mobile/src/firebase/firebase.ts [hunk 1]
```diff
@@ -241,53 +241,46 @@ export async function fetchRemoteFeatureFlags(): Promise<RemoteFeatureFlags | nu
...
-      multiTokenShowHomeBalances: flags.multiTokenShowHomeBalances.asBoolean(),
-      multiTokenUseSendFlow: flags.multiTokenUseSendFlow.asBoolean(),
-      multiTokenUseUpdatedFeed: flags.multiTokenUseUpdatedFeed.asBoolean(),
-    }
-  } else {
-    Logger.debug('No new configs were fetched from the backend.')
-    return null
+  await remoteConfig().fetchAndActivate()
+
+  const flags: FirebaseRemoteConfigTypes.ConfigValues = remoteConfig().getAll()
```
Comment (silasbw) [inline packages/mobile/src/firebase/firebase.ts:248]: total drive-by nit: many of these values aren't used as "flags".  E.g., "rewardsPercent" or "rewardsMax" or "rewardPillText" do not connote "flag".  I know the naming was here before this PR, but consider something else, like `configValues`.

File: packages/mobile/src/app/saga.ts [hunk 2]
```diff
@@ -175,11 +177,14 @@ export function* appRemoteFeatureFlagSaga() {
     const isRefreshTime = Date.now() - lastLoadTime > 60 * 60 * 1000
 
     if (isAppActive && isRefreshTime) {
-      const flags: RemoteFeatureFlags = yield call(fetchRemoteFeatureFlags)
+      const { flags }: { flags: RemoteFeatureFlags | undefined } = yield race({
```
Comment (silasbw) [inline packages/mobile/src/app/saga.ts:180]: see my annoying comment below about naming.

**Review by silasbw — COMMENTED** [review]
the small changes to the remote config handling make that code way easier to understand :man_artist:

File: packages/mobile/src/identity/verification.ts [hunk 3]
```diff
@@ -432,12 +425,6 @@ export function* requestAndRetrieveAttestations(
 
     // Check if we have a sufficient set now by fetching the new total set
     attestations = yield call(getActionableAttestations, attestationsWrapper, phoneHash, account)
-    if (shortVerificationCodesEnabled) {
-      // we only support attestation service 1.1.0 and above for short codes
-      attestations = attestations.filter(
-        (att) => !isVersionBelowMinimum(att.version, MINIMUM_VERSION_FOR_SHORT_CODES)
-      )
-    }
```
Comment (jeanregisser) [inline packages/mobile/src/identity/verification.ts:440]: Should we log if we receive attestations below the required version?

**Review by jeanregisser — APPROVED** [review]
Great! 👍

File: packages/mobile/src/identity/verification.ts (hunk 3 above)
Comment (gnardini) [reply packages/mobile/src/identity/verification.ts:440]: I think there are no attestation services with that old a version. I asked in cLabs identity channel before deleting this https://valora-app.slack.com/archives/CJXJ1EV0D/p1637346367233800 and they said this code path is no longer supported

File: packages/mobile/src/firebase/firebase.ts (hunk 1 above)
Comment (gnardini) [reply packages/mobile/src/firebase/firebase.ts:248]: Renamed :)

**Comment by ValoraQA** [comment]
Hey @MuckT  We have verified this issue using Android internal build and iOS test flight build V:1.22.2 & updated status in Gsheet:
**Gsheet Link:**
- https://docs.google.com/spreadsheets/d/1PFIydvy_ZGZVDbhrWkt6fz8mRju1jzvlNMe1k8UDMlA/edit#gid=1938857907&range=N3
**Verified On Devices:**  OnePlus 7T(11.0), Redmi Note 8(9.0)
CC: @gnardini

**Comment by ValoraQA** [comment]
Hey @MuckT We have verified this issue using Android internal build and iOS test flight build V:1.22.2 & updated status in Gsheet:
**Gsheet Link:**
- https://docs.google.com/spreadsheets/d/1PFIydvy_ZGZVDbhrWkt6fz8mRju1jzvlNMe1k8UDMlA/edit#gid=1938857907&range=R3
**Verified On Devices:**  iPhone XR (14.6), Samsung Galaxy A21s (10.0), Samsung Galaxy J5 prime(8.0)
 CC: @gnardini

**Comment by ValoraQA** [comment]
Hey @MuckT , We have verified the above issue in the latest Android Internal build V:1.22.3 & Test flight build V:1.22.3 and updated the status in test quality tool.

**Test Quality Tool Links:**
**Android 12:** https://valora.testquality.com/project/12128/plan/23022/run/153500/result/2591364
**iOS 15:** https://valora.testquality.com/project/12128/plan/23022/run/153500/result/2591224
**Devices:** Google Pixel 4a(12), iPhone 8(15.1)
CC: @gnardini
Thanks..!!

**Pull request merged**